File: src/main/java/com/dude/dms/ui/builder/tags/TagSearchResultBuilder.java
package com.dude.dms.ui.builder.tags;

import com.dude.dms.backend.data.Tag;
import com.dude.dms.backend.service.DocService;
import com.dude.dms.ui.components.dialogs.TagEditDialog;
import com.dude.dms.ui.components.search.TagSearchResult;

public final class TagSearchResultBuilder {

    private final Tag tag;

    private final TagEditDialog tagEditDialog;

    private final DocService docService;

    TagSearchResultBuilder(Tag tag, TagEditDialog tagEditDialog, DocService docService) {
        this.tag = tag;
        this.tagEditDialog = tagEditDialog;
        this.docService = docService;
    }

    public TagSearchResult build() {
        return new TagSearchResult(tag, docService.countByTag(tag), tagEditDialog);
    }
}

File: src/main/java/com/dude/dms/ui/builder/tags/TagSearchResultBuilder.kt
package com.dude.dms.ui.builder.tags

import com.dude.dms.backend.data.Tag
import com.dude.dms.backend.service.DocService
import com.dude.dms.ui.builder.Builder
import com.dude.dms.ui.components.dialogs.TagEditDialog
import com.dude.dms.ui.components.search.TagSearchResult

class TagSearchResultBuilder(
        private val tag: Tag,
        private val tagEditDialog: TagEditDialog,
        private val docService: DocService
): Builder<TagSearchResult> {

    override fun build() = TagSearchResult(tag, docService.countByTag(tag), tagEditDialog)
}